src/main/java/uk/sky/cirrus/CqlMigratorImpl.java

```diff
@@ -0,0 +1,125 @@
+ * Locking is done per client per keyspace. A client can obtain a lock on a different keyspace if no other client is holding a lock for that keyspace.
+ *
+ */
+
+public final class CqlMigratorImpl implements CqlMigrator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CqlMigratorImpl.class);
+
+    private final CassandraLockConfig lockConfig;
+
+    public CqlMigratorImpl(CassandraLockConfig lockConfig) {
+        this.lockConfig = lockConfig;
+    }
+
+    /**
+     * Allows cql migrate to be run on the command line.
+     * Set hosts, keyspace and directories using system properties.
+     */
+    public static void main(String[] args) {
+
+        final String hosts = System.getProperty("hosts");
+        final String keyspace = System.getProperty("keyspace");
+        final String directoriesProperty = System.getProperty("directories");
+        final String port = System.getProperty("port");
+
+        Preconditions.checkNotNull(hosts, "'hosts' property should be provided having value of a comma separated list of cassandra hosts");
+        Preconditions.checkNotNull(keyspace, "'keyspace' property should be provided having value of the cassandra keyspace");
+        Preconditions.checkNotNull(directoriesProperty, "'directories' property should be provided having value of the comma separated list of paths to cql files");
+
+        final Collection<Path> directories = Arrays.stream(directoriesProperty.split(","))
+                .map(Paths::get)
+                .collect(Collectors.toList());
+
+        new CqlMigratorImpl(CassandraLockConfig.builder().build())
+                .migrate(hosts.split(","), port == null ? 9042 : Integer.parseInt(port), keyspace, directories);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public void migrate(String[] hosts, int port, String keyspace, Collection<Path> directories) {
+
+        try (Cluster cluster = createCluster(hosts, port);
+             Session session = cluster.connect()) {
+            this.migrate(session, keyspace, directories);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public void migrate(Session session, String keyspace, Collection<Path> directories) {
+        final Cluster cluster = session.getCluster();
+        ClusterHealth clusterHealth = new ClusterHealth(cluster);
+        clusterHealth.check();
+
+        CassandraLockingMechanism cassandraLockingMechanism = new CassandraLockingMechanism(session, keyspace, lockConfig);
+        Lock lock = Lock.acquire(cassandraLockingMechanism, lockConfig);
+
+        LOGGER.info("Loading cql files from {}", directories);
+        CqlPaths paths = CqlPaths.create(directories);
+        final KeyspaceBootstrapper keyspaceBootstrapper = new KeyspaceBootstrapper(session, keyspace, paths);
+        final SchemaUpdates schemaUpdates = new SchemaUpdates(session, keyspace);
+        final SchemaLoader schemaLoader = new SchemaLoader(session, keyspace, schemaUpdates, paths);
```

Comment:
Any particular reason we're using final for these locals? Similar in some other places but not everywhere (e.g. #createCluster doesn't use final locals).